src/services/HttpClientService.js

```diff
@@ -0,0 +1,25 @@
+import axios from "@/axios.config";
+import { showNotification } from "@/utils/notification.js";
+
+async function fetchData(endpoint) {
+  try {
+    const response = await axios.get(endpoint);
+    if (response.status === 200) {
+      return response.data;
+    }
+  } catch (error) {
+    showNotification("Failed to fetch data from the server: ", "error");
```

Comment:
Let's also return an empty array in the catch so that it doesn't break the components. I noticed UserList is throwing an exception in the view when fetchData fails.